tests/backup/backup_share_test.go

```diff
@@ -4215,3 +4215,400 @@ var _ = Describe("{SwapShareBackup}", func() {
+							deletedUserRestoreMap[restoreName] = true
+						case api.RestoreInfo_StatusInfo_Pending:
+							pendingRestores = true
+						}
+					} else {
+						log.Warnf("restore %s with uid %s may not have been created by this testcase", restore.GetName(), restore.GetOrgId())
+					}
+				}
+				wg.Wait()
+				select {
+				case err := <-errCh:
+					return "", false, err
+				default:
+					if pendingRestores {
+						return "", true, fmt.Errorf("there are pending restores")
+					}
+					return "", false, nil
+				}
+			}
+			_, err = DoRetryWithTimeoutWithGinkgoRecover(deleteInProgressRestore, maxWaitPeriodForRestoreJobCancellation*time.Minute, restoreJobProgressRetryTime*time.Second)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Attempting to delete in progress restores"))
+			dash.VerifyFatal(len(deletedUserRestoreMap) > 0, true, "Verifying if there is at least one in progress restores")
+		})
+		Step(fmt.Sprintf("Take failed restore of backups from the user %s", infraAdminUser), func() {
+			log.InfoD(fmt.Sprintf("Taking failed restore of backups from the user %s", infraAdminUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(infraAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", infraAdminUser)
+			var wg sync.WaitGroup
+			var mu sync.RWMutex
+			createRestore := func(backupName string, restoreName string, namespace string) {
+				defer GinkgoRecover()
+				defer wg.Done()
+				namespaceMapping := map[string]string{namespace: invalidNamespace}
+				backupUID, err := Inst().Backup.GetBackupUID(nonAdminCtx, backupName, orgID)
+				log.FailOnError(err, "failed to get backup %s uid", backupName)
+				err = CreateRestoreWithUID(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, nonAdminCtx, make(map[string]string), backupUID)
+				if err != nil {
+					restoreUID, UIDError := Inst().Backup.GetRestoreUID(nonAdminCtx, restoreName, orgID)
+					log.FailOnError(UIDError, "failed to get restore %s uid", restoreName)
+					restoreInspectRequest := &api.RestoreInspectRequest{
+						Name:  restoreName,
+						OrgId: orgID,
+						Uid:   restoreUID,
+					}
+					resp, inspectError := Inst().Backup.InspectRestore(nonAdminCtx, restoreInspectRequest)
+					log.FailOnError(inspectError, "failed to inspect restore %s", backupName)
+					if resp.GetRestore().GetStatus().GetStatus() != api.RestoreInfo_StatusInfo_Failed {
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of restore %s of backup %s", restoreName, backupName))
+					}
+				}
+				mu.Lock()
+				defer mu.Unlock()
+				userFailedRestoreNames = append(userFailedRestoreNames, restoreName)
+			}
+			for backupName, namespace := range userSuccessfulBackupMap {
+				wg.Add(1)
+				restoreName := fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
+				go createRestore(backupName, restoreName, namespace)
+			}
+			wg.Wait()
+			log.Infof("The list of failed user restores taken are: %v", userFailedRestoreNames)
+		})
```

Comment:
I might be missing a step to verify restore by non-admin user is listed from admin pages while restore is in failed state. @ak-px

Reply:
@kphalgun-px , owner id for restore is not supported in enumerate options, you can drop listing for restore deletion steps

Reply:
Done. Used GetAllRestoreNamesByOwnerID in topic/krishna-px/PA-1528-v7.